### PR TITLE
lint against non-kebab-cased enum values

### LINKED ISSUE #544: consider linting against camelCase enum values
`~enums~` are written with inconsistent casing within and across specs. https://github.com/tc39/ecmarkup/pull/539 will change the styling to smallcaps, which can look a little bit weird with capital letters, so that might be a good impetus to start enforcing `~kebab-case~` enum values everywhere.

cc @tc39/ecma402-editors because this would entail a change in 402 (which has some kebab-case, some camelCase, and some enum values with spaces).

### PR DESCRIPTION
**Pull request by bakkot**
Fixes #544.

### CODE REVIEW

File: src/lint/rules/enum-casing.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+/*
+Checks that ~enum-values~ are kebab-cased.
+*/
+export default function (report: Reporter, step: OrderedListItemNode, algorithmSource: string) {
+  for (const item of step.contents) {
+    if (item.name !== 'tilde' || item.contents.length !== 1 || item.contents[0].name !== 'text') {
+      continue;
+    }
+    const text = item.contents[0];
+    if (/[A-Z ]/.test(text.contents)) {
```
Comment (michaelficarra) [inline src/lint/rules/enum-casing.ts:16]:
```suggestion
    if (!/^[\p{Ll}\d]+(?:-[\p{Ll}\d]+)*$/v.test(text.contents)) {
```

File: src/lint/rules/enum-casing.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      const location = offsetToLineAndColumn(algorithmSource, text.location.start.offset);
+      report({
+        ruleId,
+        message: 'enum values should be kebab-cased',
```
Comment (michaelficarra) [inline src/lint/rules/enum-casing.ts:20]:
```suggestion
        message: 'enum values should be lowercase and kebab-cased',
```

**Review by michaelficarra — COMMENTED** [review]
LGTM otherwise

File: src/lint/rules/enum-casing.ts (hunk 2 above)
Comment (bakkot) [reply src/lint/rules/enum-casing.ts:20]: Wait what? Doesn't kebab-case imply lowercase?

Comment (michaelficarra) [reply src/lint/rules/enum-casing.ts:20]: `THIS-IS-STILL-KEBAB`

**Review by michaelficarra — APPROVED** [review]

**Pull request merged**